src/client/app/containers/BarChartContainer.ts

```diff
@@ -45,87 +51,77 @@ function mapStateToProps(state: State) {
 			const readingsData = byGroupID[timeInterval.toString()][barDuration.toISOString()];
 			if (readingsData !== undefined && readingsData.readings !== undefined && !readingsData.isFetching) {
 				const label = state.groups.byGroupID[groupID].name;
-				const color = getGraphColor(label);
+				const xData: string[] = [];
+				const yData: number[] = [];
+				const hoverText: string[] = [];
+				const readings = _.orderBy(readingsData.readings, ['startTimestamp'], ['asc']);
+				readings.forEach(barReading => {
+					const timeReading: string = `${moment(barReading.startTimestamp).format('MMM DD, YYYY')} - ${moment(barReading.endTimestamp).format('MMM DD, YYYY')}`;
+					xData.push(timeReading);
+					yData.push(barReading.reading);
+					hoverText.push(`<b> ${timeReading} </b> <br> ${label}: ${barReading.reading} kW`);
+				});
+
 				datasets.push({
-					label,
-					data: readingsData.readings.map(barReading => barReading.reading),
-					backgroundColor: color,
-					hoverBackgroundColor: color
+					name: label,
+					x: xData,
+					y: yData,
+					text: hoverText,
+					hoverinfo: 'text',
+					type: 'bar',
+					marker: {color: getGraphColor(label)}
 				});
-				// Add only the unique time intervals to the label set
-				for (const barReading of readingsData.readings) {
-					labelsSet.add(
-						`${moment(barReading.startTimestamp).format('MMM DD, YYYY')} - ${moment(barReading.endTimestamp).format('MMM DD, YYYY')}`);
-				}
 			}
 		}
 	}
 
-	// Converts the label set into an array for Chart.js and sorts the labels based on the first date of the time interval
-	const labels = Array.from(labelsSet).sort((x, y) => {
-		const t1 = moment(x.split(' - ')[0], 'MMM DD, YYYY').format('x');
-		const t2 = moment(y.split(' - ')[0], 'MMM DD, YYYY').format('x');
-		return +(t1) - +(t2);
-	});
 
-	const options = {
-		animation: {
-			duration: 0
+	// Customize the layout of the plot
+	const layout: any = {
+		barmode: 'group',
+		bargap: 0.2,
+		bargroupgap: 0.1,
+		autozise: true,
+		showlegend: true,
+		legend: {
+			x: 0,
+			y: 1.1,
+			orientation: 'h',
 		},
-		elements: {
-			point: {
-				radius: 0
-			}
+		yaxis: {
+			title: 'kW',
+			showgrid: true,
+			gridcolor: '#ddd'
 		},
-		scales: {
-			xAxes: [{
-				stacked: state.graph.barStacking,
-				gridLines: {
-					display: true
-				}
-			}],
-			yAxes: [{
-				stacked: state.graph.barStacking,
-				scaleLabel: {
-					display: true,
-					labelString: 'kW'
-				},
-				ticks: {
-					min: 0
-				}
-			}]
+		xaxis: {
+			showgrid: true,
+			gridcolor: '#ddd',
+			tickfont: {
+				size: 10
+			},
+			tickangle: -45,
+			autotick: true,
+			nticks: 10,
```

Comment:
<https://plot.ly/javascript/reference/#scattergl-marker-colorbar-tickmode> does not seem to make it clear if tickmode is auto (at least not from my very modest checking). nticks only has impact in this case. Should we set tickmode?

Reply:
Upon rereading, it states for the other two cases what is the default so it seems it is likely okay in this case. Since working, I am going to close this.